src/main/java/build/buildfarm/server/BuildFarmServer.java

```diff
@@ -123,8 +112,9 @@ public BuildFarmServer(String session, ServerBuilder<?> serverBuilder, BuildFarm
         .addService(new OperationQueueService(instances))
         .addService(new OperationsService(instances));
 
-    if (!tokens.isEmpty()) {
-        serverBuilder.intercept(new StaticAuthInterceptor(tokens));
+    ServerInterceptor authHeaderInterceptor = getAuthInterceptor();
+    if (authHeaderInterceptor != null) {
```

Comment:
i'd like the server to not load if `authHeaderInterceptor` is null.

Reply:
Then I'll have to add a command-line flag to enable that behavior. This same code is used for local and dev cluster development where it should load in that case.

Reply:
we should load a JWT tokens in those cases too (different ones, obviously).
we do that in dev for other stuff (django secrets, github, etc...)

at any case, default behavior should be use tokens unless there is an explicit option to disable it.

Reply:
Also, I accounted for this already because `getAuthInterceptor` will throw if it cannot read the key file. The null is only returned if the environment variable is not set. Is that sufficient?

Reply:
The integration test is structured to use BuildFarmServer and requires it to be unauthenticated. I didn't touch that test (and am wary to do so) in this change. To address your concern, I've restructured the interceptor setup so it is clear that it must succeed if auth is enabled.

Reply:
To clarify, we run full tests in the buidfarm image builder script.